cpp/include/raft/core/math.hpp

```diff
@@ -91,7 +96,14 @@ RAFT_INLINE_FUNCTION auto atanh(T x)
 
 /** Cosine */
 template <typename T>
-RAFT_INLINE_FUNCTION auto cos(T x)
+RAFT_INLINE_FUNCTION
+#if _RAFT_HAS_CUDA
+  typename std::enable_if<!std::is_same<T, __half>::value && !std::is_same<T, nv_bfloat16>::value,
+                          T>::type
```

Comment:
Nitpick: can you use C++17 constructs for readability? `enable_if_t` and `is_same_v` (everywhere, not just this line)